platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/thermal_manager.py

```diff
@@ -9,11 +9,13 @@ def __init__(self, polling_time = 30.0):
         self.__thermals = None
         self.__timer = None
         self.__chassis = None
+        self.__running = False
         
     def start(self):
-        self.work()
-        self.__timer = Timer(self.__polling_thermal_time, self.start)
-        self.__timer.start()
+        if self.__running == True:
```

Comment:
should not be `if not self.__running:` ?

Reply:
In this case it will be simple return without any action